jinja2/react_base.html

```diff
@@ -104,33 +104,35 @@
 
   {%- block contrib_popup %}{# Inject contributions popup where needed #}{% endblock %}
 
-  <!-- Footer -->
-  <footer id="nav-footer" class="nav-footer">
-    <div class="center">
-        {% include "includes/footer/mdn-footer.html" %}
-
-        {% include "includes/footer/moz-footer.html" %}
-
-        {% block lang_switcher %}
-          {% include "includes/lang_switcher.html" %}
-        {% endblock %}
-
-        {% block footer_copyright %}
-          <ul class="footer-tos">
-            <li><a href="https://www.mozilla.org/about/legal/terms/mozilla">{{ _('Terms') }}</a></li>
-            <li><a href="https://www.mozilla.org/privacy/websites/">{{ _('Privacy') }}</a></li>
-            <li><a href="https://www.mozilla.org/privacy/websites/#cookies">{{ _('Cookies') }}</a></li>
-          </ul>
-
-          <div id="license" class="contentinfo">
-            {% trans copyright_url='/en-US/docs/MDN/About#Copyrights_and_licenses', thisyear=thisyear() %}
-              <p>&copy; 2005-{{ thisyear }} Mozilla and individual contributors.</p>
-              <p>Content is available under <a href="{{ copyright_url }}">these licenses</a>.</p>
-            {% endtrans %}
-          </div>
-        {% endblock %}
-    </div>
-  </footer>
+  {% block footer %}
+    <!-- Footer -->
+    <footer id="nav-footer" class="nav-footer">
+      <div class="center">
+          {% include "includes/footer/mdn-footer.html" %}
+
+          {% include "includes/footer/moz-footer.html" %}
+
+          {% block lang_switcher %}
+            {% include "includes/lang_switcher.html" %}
+          {% endblock %}
+
+          {% block footer_copyright %}
+            <ul class="footer-tos">
+              <li><a href="https://www.mozilla.org/about/legal/terms/mozilla">{{ _('Terms') }}</a></li>
+              <li><a href="https://www.mozilla.org/privacy/websites/">{{ _('Privacy') }}</a></li>
+              <li><a href="https://www.mozilla.org/privacy/websites/#cookies">{{ _('Cookies') }}</a></li>
+            </ul>
+
+            <div id="license" class="contentinfo">
+              {% trans copyright_url='/en-US/docs/MDN/About#Copyrights_and_licenses', thisyear=thisyear() %}
+                <p>&copy; 2005-{{ thisyear }} Mozilla and individual contributors.</p>
+                <p>Content is available under <a href="{{ copyright_url }}">these licenses</a>.</p>
+              {% endtrans %}
+            </div>
+          {% endblock %}
+      </div>
+    </footer>
+  {% endblock %}
```

Comment:
I still cannot understand why this change is necessary? 
As far as I can tell, the block of code was put in a `{% block footer %} ... {% endblock %}` But is there any reason to do it?

Reply:
If we don't have this, it renders two footers. So this allows us to override this footer in `payments/index.html`:

<img width="1036" alt="two-footers" src="https://user-images.githubusercontent.com/650/76352585-af854100-62e5-11ea-8a98-c52e46d5649a.png">

Reply:
I could remove the React Footer and leave the one here, but when do we know which one to use? The React Footer is only used in documents right now.

Reply:
I give up on all this footer stuff. I think you an @schalkneethling should push back on the influx of feature additions and convince management that you have to refactor this mess up. 
As mentioned in Slack, we have **4** `<html ...` tags, when it should be 2 (one for the Wiki and one for the read-only site). 
```
▶ rg '<html ' | rg -v jquery | rg -v ckeditor | rg -v tests | rg -v '\.ltxt'
jinja2/base.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
jinja2/react_base.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
kuma/wiki/jinja2/wiki/react_document.html:<html lang="{{ document_data['hrefLang'] }}" dir="{{ DIR }}">
kuma/users/jinja2/socialaccount/base_signup.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
```
And **4** `<footer` tags when there should be 2. 
```
▶ rg '<footer' | rg -v jquery | rg -v ckeditor | rg -v tests | rg -v '\.ltxt' | rg -v l10n.js
jinja2/base.html:    <footer id="nav-footer" class="nav-footer">
jinja2/react_base.html:    <footer id="nav-footer" class="nav-footer">
kuma/javascript/src/footer.jsx:        <footer id="nav-footer" className="nav-footer">
kuma/users/jinja2/socialaccount/base_signup.html:    <footer id="nav-footer" class="nav-footer">
```

Reply:
Ah ok, I am on the same page here now. The change was indeed simple to wrap the `footer` HTML inside a `footer` Jinja block. This is required because `payments/index.html` extends `react_base.html`. If the `footer` is not wrapped in a `block` there is no way for us to override the `block` inside `payments/index.html`

The result would be two footers. One from React and one from the extended `react_base.html`. The reason `search/react.html` did not run into this scenario first is that it does not use the React footer but the one inherited from `react_base.html`

This means that as long as we have pages that need to extend `react_base.html` the following will always be the template used by said page:

```
{% extends "react_base.html" %}

{% block title %}{{ page_title(_('Become a monthly supporter')) }}{% endblock %}

{% block document_head %}
   {#
        This page doesn't need to fetch data, so we pass `True`
        as an argument to indicate that we have the data we need, which
        allows the component to render in the server.
    #}
    {{ render_react("SPA", request.LANGUAGE_CODE, request.get_full_path(), True)|safe }}
{% endblock %}

{% block content %}{% endblock %}

{% block footer %}{% endblock %}
```

SPA is probably not the best naming scheme anymore inside Kuma anyway. RPC(React Page Component) might be more appropriate :)

Reply:
With that said, if all pages that extend `react_base.html` do not need the footer any longer, we can remove the footer from `react_base.html` and in so doing, the need for the Jinja `footer block`.

Currently, though, three pages extend it:

1. `landing/react_homepage.html`
2. `payments/index.html`
3. `search/react.html`

Numbers 2 and 3 do not need the footer but, until the homepage is converted to React, it will still need it.

Reply:
Opened up https://github.com/mdn/kuma/issues/6678 so we don't loose track of this.